### PR TITLE
Reorder test script for jenkins ci

### PR DESCRIPTION
**Pull request by reachjainrahul**
All the scripts will take command line arguments.
Also, move common installation in a separate file.

Signed-off-by: Rahul Jain <rahulj@vmware.com>

### CODE REVIEW

**Comment by reachjainrahul** [comment]
/nephe-test-e2e-eks, /nephe-test-e2e-aws,  /nephe-test-e2e-azure

File: ci/jenkins/nephe-ci.sh [hunk 1]
```diff
@@ -151,33 +151,36 @@ ssh -i id_rsa ubuntu@${ip_addr} "mkdir ~/nephe"
...
 
 function cleanup_testbed() {
-  echo "=== retrieve logs ==="
-  scp -r -i id_rsa ubuntu@${ip_addr}:~/logs ${OLDPWD}
+    echo "=== retrieve logs ==="
+    scp -r -i id_rsa ubuntu@${ip_addr}:~/logs ${OLDPWD}
 
-  echo "=== cleanup vm ==="
-  ./destroy.sh "${testbed_name}" "${goVcPassword}"
+    echo "=== cleanup vm ==="
```
Comment (Anandkumar26) [inline ci/jenkins/nephe-ci.sh:157]:
```suggestion
    echo "=== Cleanup VM ==="
```

File: ci/jenkins/nephe-ci.sh [hunk 2]
```diff
@@ -151,33 +151,36 @@ ssh -i id_rsa ubuntu@${ip_addr} "mkdir ~/nephe"
 scp -r -i id_rsa ${OLDPWD}/* ubuntu@${ip_addr}:~/nephe
 
 function cleanup_testbed() {
-  echo "=== retrieve logs ==="
-  scp -r -i id_rsa ubuntu@${ip_addr}:~/logs ${OLDPWD}
+    echo "=== retrieve logs ==="
```
Comment (Anandkumar26) [inline ci/jenkins/nephe-ci.sh:154]:
```suggestion
    echo "=== Retrieve logs ==="
```

File: ci/jenkins/scripts/install-common.sh [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+GO_VERSION=1.19
+
+function install_common_packages() {
+    echo "Installing Kubectl"
+    curl -LO https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl
+    chmod +x ./kubectl && sudo mv ./kubectl /usr/local/bin/kubectl
+
+    sudo apt-get install -y pv bzip2 jq unzip
+
+    echo "Installing Terraform"
```
Comment (Anandkumar26) [inline ci/jenkins/scripts/install-common.sh:31]:
```suggestion
    echo "Installing Terraform version ${TERRAFORM_VERSION}"
```

File: ci/jenkins/scripts/install-common.sh [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    echo "Installing Go ${GO_VERSION}"
+    curl -LO https://golang.org/dl/go${GO_VERSION}.linux-amd64.tar.gz
+    sudo rm -rf /usr/local/go && sudo tar -zxf go${GO_VERSION}.linux-amd64.tar.gz -C /usr/local/
+    rm go${GO_VERSION}.linux-amd64.tar.gz
+    export PATH=$PATH:/usr/local/go/bin
+}
+
+function install_kind() {
+    echo "Installing Kind"
```
Comment (Anandkumar26) [inline ci/jenkins/scripts/install-common.sh:45]:
```suggestion
    echo "Installing Kind version ${KIND_VERSION}"
```

File: ci/jenkins/scripts/test-aks.sh [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+set -e
+
+function echoerr {
+    >&2 echo "$@"
+}
+
+_usage="Usage: $0 [arguments]
+
+Setup and Run integration test on AKS cluster with Azure VMs.
```
Comment (Anandkumar26) [inline ci/jenkins/scripts/test-aks.sh:25]:
```suggestion
Setup and Run integration tests on AKS cluster with Azure VMs.
```

File: ci/jenkins/scripts/test-aks.sh [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+fi
+
+source $(dirname "${BASH_SOURCE[0]}")/install-common.sh
+install_common_packages
+
+echo "Building Nephe Docker image"
+make build
+
+echo "Installing Azure CLI"
+curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
```
Comment (Anandkumar26) [inline ci/jenkins/scripts/test-aks.sh:101]: Are we dependent on any particular version of Azure CLI?

File: ci/jenkins/scripts/test-aks.sh [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+    *)    # unknown option
+    echoerr "Unknown option $1"
+    print_help
+    exit 1
+    ;;
+esac
+done
+
+if [ -z "$TF_VAR_azure_client_subscription_id" ] || [ -z "$TF_VAR_azure_client_id" ] || [ -z "$TF_VAR_azure_client_tenant_id" ] || [ -z "$TF_VAR_azure_client_secret" ]; then
+    echo "Azure credentials must be set."
```
Comment (Anandkumar26) [inline ci/jenkins/scripts/test-aks.sh:89]: Why not use echoerr for all the error related logs?

File: ci/jenkins/scripts/test-aws.sh [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+# limitations under the License.
+
+set -e
+
+function echoerr {
+    >&2 echo "$@"
+}
+
+_usage="Usage: $0 [arguments]
+Setup and run integration test on Kind cluster with AWS VMs.
```
Comment (Anandkumar26) [inline ci/jenkins/scripts/test-aws.sh:24]:
```suggestion
Setup and run integration tests on Kind cluster with AWS VMs.
```

File: ci/jenkins/scripts/test-aws.sh [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+echo "Building Nephe Docker image"
+make build
+
+install_kind
+pull_docker_images
+
+# Tag locally built nephe image
+docker tag antrea/nephe:latest projects.registry.vmware.com/antrea/nephe:latest
+
+echo "Creating kind cluster"
```
Comment (Anandkumar26) [inline ci/jenkins/scripts/test-aws.sh:100]:
```suggestion
echo "Creating Kind cluster"
```

File: ci/jenkins/scripts/test-azure.sh [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+echo "Building Nephe Docker image"
+make build
+
+install_kind
+pull_docker_images
+
+# Tag locally built image
+docker tag antrea/nephe:latest projects.registry.vmware.com/antrea/nephe:latest
+
+echo "Creating kind cluster"
```
Comment (Anandkumar26) [inline ci/jenkins/scripts/test-azure.sh:106]:
```suggestion
echo "Creating Kind cluster"
```

File: ci/jenkins/scripts/test-azure.sh [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+
+set -e
+
+function echoerr {
+    >&2 echo "$@"
+}
+
+_usage="Usage: $0 [arguments]
+
+Setup and Run integration test on Kind cluster with Azure VMs.
```
Comment (Anandkumar26) [inline ci/jenkins/scripts/test-azure.sh:25]:
```suggestion
Setup and Run integration tests on Kind cluster with Azure VMs.
```

File: ci/jenkins/scripts/test-azure.sh [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+    *)    # unknown option
+    echoerr "Unknown option $1"
+    print_help
+    exit 1
+    ;;
+esac
+done
+
+if [ -z "$TF_VAR_azure_client_subscription_id" ] || [ -z "$TF_VAR_azure_client_id" ] || [ -z "$TF_VAR_azure_client_tenant_id" ] || [ -z "$TF_VAR_azure_client_secret" ]; then
+    echo "Azure credentials must be set."
```
Comment (Anandkumar26) [inline ci/jenkins/scripts/test-azure.sh:89]:
```suggestion
    echoerr "Azure credentials must be set."
```

File: ci/jenkins/scripts/test-eks.sh [hunk 13]
```diff
@@ -0,0 +1,155 @@
...
+# limitations under the License.
+
+set -e
+
+function echoerr {
+    >&2 echo "$@"
+}
+
+_usage="Usage: $0 [arguments]
+Setup and run integration test on EKS cluster with AWS VMs.
```
Comment (Anandkumar26) [inline ci/jenkins/scripts/test-eks.sh:24]:
```suggestion
Setup and run integration tests on EKS cluster with AWS VMs.
```

**Review by Anandkumar26 — COMMENTED** [review]
Looks good to me. 
Some minor comments w.r.t logging.

File: ci/jenkins/scripts/test-aks.sh (hunk 6 above)
Comment (reachjainrahul) [reply ci/jenkins/scripts/test-aks.sh:101]: We will use the latest cli version.

File: ci/jenkins/scripts/test-aks.sh (hunk 7 above)
Comment (reachjainrahul) [reply ci/jenkins/scripts/test-aks.sh:89]: thats only for usage and help.. In normal case, echo is prefered

**Review by Anandkumar26 — APPROVED** [review]
/LGTM

**Pull request merged**